tests/platform_tests/api/conftest.py

```diff
@@ -10,11 +11,12 @@
 IPTABLES_PREPEND_RULE_CMD = 'iptables -I INPUT 1 -p tcp -m tcp --dport {} -j ACCEPT'.format(SERVER_PORT)
 IPTABLES_DELETE_RULE_CMD = 'iptables -D INPUT -p tcp -m tcp --dport {} -j ACCEPT'.format(SERVER_PORT)
 
+PLATFORM_API_SERVER_MAX_TRY = 3
 
-@pytest.fixture(scope='function')
-def start_platform_api_service(duthosts, enum_rand_one_per_hwsku_hostname, localhost, request):
-    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
-    dut_ip = duthost.mgmt_ip
+logger = logging.getLogger(__name__)
+
+
+def platform_api_service_start_helper(duthost, dut_ip, localhost, request):
 
```

Comment:
@vdahiya12 can we use [wait_until](https://github.com/sonic-net/sonic-mgmt/blob/master/tests/common/utilities.py#L106)